### ISSUE TITLE
Use of std::sprintf causes deprecation warnings on new Mac OS (Ventura)

### ISSUE THREAD

**Issue by phillip-keldenich** [doctest/doctest#724]
## Description
The use of std::sprintf in doctest headers (used to convert integers to strings) causes warnings (because of 'deprecation') when compiling with the AppleClang compiler shipped with the new XCode version on MacOS Ventura. Any code that uses std::sprintf or sprintf will cause a warning on that compiler/stdio.h header, and suggests to use std::snprintf instead.
This is, of course, overzealous in the case of number -> string conversion (unless the platform has unusually wide integers).

### Steps to reproduce
Just compile a very small test example on that compiler where doctest has to turn a number into a string, e.g.,
```
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest/doctest.h"
#include <vector>

TEST_CASE("foo") {
	std::vector<double> foo{1.2, 1.3, 1.4};
	CHECK(foo == std::vector<double>{1.2, 1.3, 1.4});
}
```

### Extra information
* doctest version: 2.4.8
* Operating System: MacOS Ventura
* Compiler+version: Apple clang version 14.0.0 (clang-1400.0.29.202)

**Comment by LevyMatan** [comment]
It looks like it was solved in 2.4.9.

**Comment by phillip-keldenich** [comment]
Indeed, it was. So this issue can be closed.
